src/module/actor/character/data/sheet.ts

```diff
@@ -34,18 +34,25 @@ export interface CraftingEntriesSheetData {
     alchemical: {
         entries: CraftingEntry[];
         totalReagentCost: number;
-        infusedReagents: {
-            value: number;
-            max: number;
-        };
     };
 }
 
 interface CraftingSheetData {
     noCost: boolean;
     hasQuickAlchemy: boolean;
+    hasQuickTincture: boolean;
     knownFormulas: Record<number, CraftingFormula[]>;
     entries: CraftingEntriesSheetData;
+    resources: {
+        infusedReagents: {
+            value: number;
+            max: number;
+        };
+        versatileVials: {
+            value: number;
+            max: number;
+        };
+    };
```

Comment:
This is isn't going to work. We're not hard-coding every advanced-alchemy-like feature ever published and to be published.

Reply:
Attempt made to make this more generic.

Should I also extend this to crafting resources not being strictly typed?